lubot.js

```diff
@@ -175,23 +175,40 @@ bot.helpers.utils = {
     return false;
   },
   stripUpKarma: function(text) {
-    var re = new RegExp("([A-Za-z0-9]{1,})(?=[ >:]*\\+\\+)");
-    var matches = re.exec(text);
-    if (matches) {
-      return(matches[1]);
-    }
-    return false;
+    return bot.helpers.utils.stripKarma(text, 'up');
   },
   stripDownKarma: function(text) {
-    var re = new RegExp("([A-Za-z0-9]{1,})(?=[ >:]*\-\-)");
-    var matches = re.exec(text);
+    return bot.helpers.utils.stripKarma(text, 'down');
+  },
+  stripKarma: function(text, direction) {
+    var re = new RegExp('^(.+)(?=[ >:]*' + (direction === 'down' ? '--' : '\\+\\+') + '$)', 'i'),
```

Comment:
Why do we need this case insensitive? We're not matching anything case sensitive, so I think that might be unnecessary, no?